File: app/src/main/java/com/example/mobile/julfani/tubes/entity/SleepRoomDatabase.java
package com.example.mobile.julfani.tubes.entity;

import android.content.Context;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;

@Database(entities = {Sleep.class}, version = 1, exportSchema = false)
public abstract class SleepRoomDatabase extends RoomDatabase {
    public abstract SleepDao sleepDao();
    private static volatile SleepRoomDatabase INSTANCE;

    static SleepRoomDatabase getDatabase(final Context context){
        if (INSTANCE == null){
            synchronized (SleepRoomDatabase.class){
                if(INSTANCE == null){
                    INSTANCE = Room.databaseBuilder(context.getApplicationContext(), SleepRoomDatabase.class, "sleep_database").build();
                }
            }
        }
        return INSTANCE;
    }
}

File: app/src/main/java/com/nursyah/bettersleep/entity/SleepRoomDatabase.kt
package com.nursyah.bettersleep.entity

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase

@Database(entities = [Sleep::class], version = 1, exportSchema = false)
abstract class SleepRoomDatabase : RoomDatabase() {
    abstract fun sleepDao(): SleepDao?

    companion object {
        @Volatile
        private var INSTANCE: SleepRoomDatabase? = null

        fun getDatabase(context: Context): SleepRoomDatabase? {
            return INSTANCE?: synchronized(this) {
                        val instance = Room.databaseBuilder(context.applicationContext, SleepRoomDatabase::class.java, "sleep_database").build()
                        INSTANCE = instance
                        instance
            }
        }
    }
}